### ISSUE TITLE
XCronBundle, stored cronjob action documents aren't deleted from mongo

### ISSUE THREAD

**Issue by AranNomante** [bluelibs/bluelibs#337]
**Describe the bug**
Cron action documents aren't deleted from mongo so as time passes the document count gets inflated.

**To Reproduce**
1. Set up a cron scheduled to run every 1-10 seconds
2. Observe the mongo for few days
3. Crons will only go up in document count
4. We have 253k documents & 53k documents on few deployments even though they haven't been running more than a month with fresh datasets. 

**Expected behavior**
You would expect unrelevant or cron actions older than the last action to be deleted permanently. However, even the oldest ones aren't even soft deleted.

**Screenshots**
taken 2024-18-04 10:10 TRT time
![image](https://github.com/bluelibs/bluelibs/assets/44326591/cbfb45aa-5b08-4562-bfb0-d33cea5736c4)

**Server:**
 - OS: possibly alpine

**Comment by theodorDiaconu** [comment]
Makes sense, si Crons are part of the X bundle, then it means we assume Mongo, and we can add a configuration to the bundle:
new XCronBundle({ deleteScheduledCronjobsOlderThan: '7d', });  // default it like that but leave option to opt-out

And we add a daily cronjob that simply deletes it based on the config.

**Comment by theodorDiaconu** [comment]
@AranNomante actually 

```

## Persistence

By default jobs are saved into the database, to bypass this use:

```ts
cronService.add({
  // ...
  persist: false,
})
```

And if you want, you can clean them up from the Crons collection yourself. 
import { CronsCollection } from "@bluelibs/x-cron-bundle"
